src/shared/behaviors/Validatable.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright (c) 2015 MediaMath Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
+
+*/
+(function (scope) {
+	
+	scope.Validate = {
+
+		listeners: {
+			"value-changed" : "_validateOnUpdate"
+		},
+
+		properties: {
+			error: {
+				type: Boolean,
+				value: false
+			},
+			validation: {
+				type: String,
+				value: false,
+				observer: "_validationChanged"
+			}
+		},
+
+		// common: 
+		email: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+		alpha: /^\w+$/,
+		inter: /^\d+$/,
+		floater: /^\d*[.]\d+$/,
+		whitespace: /\s/,
+
+		// items to validate against:
+		validators: null,
+
+		_validationChanged: function(newVal, oldVal) {
+			if (newVal) {
+				this.validators = newVal.replace(/\s/g, '').split("|");
+			}
+		},
+
+		_isValid: function(validator) {
+			switch(validator) {
+				case "checked":
```

Comment:
if you move the switch to an object hash with named methods ('checked':function() { return this.value===true} then you could chain a map with the filter. The added benefit of this is that the object could be exposed publicly so people could add their own validators and re-use the overall system without throwing away validator completely.